### PR TITLE
Package SEO improves, BTable updates, and Docs updates

### PR DESCRIPTION
**Pull request by devhus**
Package SEO improves, Please read this issue to understand why the package info needs to be updated.
https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1261

BTable updates:
- feat(BTable): updated table field formatter to a reusable type
- feat(BTable): corrected field object formatter type
- feat(BTable): corrected item rendering logic, allowing easier updates to the renderItem function which can now include more logic for rendering besides formatting
- BREAKING CHANGE(BTable): Updated the items processing logic, Now items are fully reactive and can be updated directly through the item's cell slot.
- feat(BTable): removed items cloning when emitting to the filtered handler to deliver reactive items
- feat(BTable): sortBy & sortDesc now can be used to pre-sort the table  items without making any fields sortable
- feat(BTable): the selection box is now can be modified through CSS uisng the ``.b-table-selection-icon`` and ``.selected`` selectors
- feat(BTable): added tbodyTrClass prop to update the tbody > rows styles based on function results
- feat(BTable): the sort-by prop now has a shell value to be more flexible and can be used as a v-model, read-only prop, or don't be passed at all

Note that items cloning is partly removed and that should improve the component's performance when displaying a large number of rows that need to be sorted or filtered.

This PR should solve these issues:
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1261
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/958
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1061
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1251 (partly)
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1253
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1269
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1270
- https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/995

### CODE REVIEW

File: packages/bootstrap-vue-next/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "bootstrap-vue-next",
-  "displayName": "BootstrapVueNext",
-  "description": "Early (but lovely) implementation of Vue 3, Bootstrap 5 and Typescript",
+  "displayName": "BootstrapVue3",
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/package.json:3]: Do not change the display name. It will stay BootstrapVueNext

File: packages/bootstrap-vue-next/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+  "description": "Bootstrap Vue 3 library is an early and lovely for Vue 3 & Nuxt 3 based on Bootstrap 5 and Typescript.",
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/package.json:4]: Modification of the description is "fine", but review above.

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
+import {computed, ref, type Ref} from 'vue'
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table-items.ts:1]: This file name should be camelCase

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+import type {BTableProps} from './table'
+import type {BTableSortCompare, TableField, TableFieldObject, TableItem} from 'src/types'
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table-items.ts:3]: You cannot import from src/ use relative file paths

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+import type {TableFieldObjectFormatter} from 'src/types/TableFieldObject'
+
+export function useTableItems(
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table-items.ts:6]: arrow functions  > functions

Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table-items.ts:6]: I'd prefer this code to simply be in the actual BTable file. It may look bloated, but I think it outweighs the drawback of code being hidden from first view. I similarly don't like how BFormInput has a special composable, but in that case, it is used in multiple places. This is not.

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  tableProps: BTableProps,
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table-items.ts:7]: In addition to the above, I'd prefer BTableProps to not be declared in a separate file. (Review below)

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+  flags: Record<string, Ref<boolean>>,
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table-items.ts:8]: It may be better to use `Record<string, MaybeRefOrGetter<boolean>>`, but you will need to handle it differently... However this may be a non issue

File: packages/bootstrap-vue-next/src/components/BTable/table.d.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  Booleanish,
+  Breakpoint,
+  BTableProvider,
+  BTableSortCompare,
+  ColorVariant,
+  VerticalAlign,
+} from '../../types'
+
+export interface BTableProps {
+  align?: VerticalAlign
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table.d.ts:11]: As mentioned above, I'd prefer the composable to be moved back into the BTable file alone. Along with this, this type can be moved back into the position it was before. I am preferring to not have different declaration files anymore.

File: packages/bootstrap-vue-next/src/components/BTable/table.d.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import type {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/table.d.ts:1]: In addition to this, even though this file should be removed, you are not supposed to write .d.ts files. Review https://www.youtube.com/watch?v=zu-EgnbmcLY

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue [hunk 10]
```diff
@@ -362,8 +361,7 @@ filteredHandler.value = async (items) => {
     await callItemsProvider()
     return
   }
-  const clone = await cloneDeepAsync(items)
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTable.vue]: I agree, the clone deep didn't serve much purpose. You can likely remove the cloneDeep function and the subsequent cloneDeepAsync. But you'd want to check first.

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts (hunk 4 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts:3]: Yeah, long time not working on the library i forgot that part xD

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts (hunk 5 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts:6]: I wanted to have a clean, more readable code. Having the logic of the items handling as composable would be more maintainable if we need to change anything related to the items processing but not to the interface. Think about it and let me know if you still wanna merge interface and items logic into one large file.

Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts:6]: In my opinion, moving one monstrous ugly code from one place to another, doesn't fix the problem of it being monstrous and ugly. Not to mention, I'm not a fan of the idea of having the parameter be "the entire props object", things get messy, reactivity wise as it doesn't quite fit the expected flow (at least in my mind)

Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts:6]: Hmmm, Idk it does feel better to have the items processing code in a standalone file. Keep in mind that most of the table-items.ts code was already existing in itemHelper.ts file which is also a standalone file.

File: packages/bootstrap-vue-next/package.json (hunk 1 above)
Comment (devhus) [reply packages/bootstrap-vue-next/package.json:3]: Np

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts (hunk 7 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts:8]: It will not be an issue since all of the given flags are Ref-like anyway, if there are getters added to the flags we can change the types for sure. But for now, there is no need to add any extra validation for the type.

File: packages/bootstrap-vue-next/src/components/BTable/table.d.ts (hunk 9 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table.d.ts:1]: Good to know, thanks for the tip

File: packages/bootstrap-vue-next/src/components/BTable/table.d.ts (hunk 8 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table.d.ts:11]: The logic used within the composable has been always outside the BTable file. So do you want to remove the idea of having external files for logic processing, or you just don't like to have the props declaration outside the .vue file?

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue (hunk 10 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/BTable.vue]: I have checked with a few manual tests, Everything seems to be working as it was except now items are reactive everywhere.

File: packages/bootstrap-vue-next/src/components/BTable/table-items.ts (hunk 5 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts:6]: So, this file is simply a reactive alternative for itemHelper.ts file.  https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/1262/commits/c6e004fc41b5678730db0b18eaf7b5e55880dc4c#diff-ef682efae412cee917e8dc66838252a8c9d3189da77868f1aeab58e8b078f4df

If the problem with moving the props type outside BTable.vue file let me know and i will handle it.

Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-items.ts]: I've submitted a few updates to the PR, let me know if this is enough or not to continue working on other issues

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue (hunk 10 above)
Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BTable/BTable.vue]: Typically the setter getters and proxies are destroyed during the serialization process, which breaks reactivity.

File: packages/bootstrap-vue-next/src/components/BTable/table.d.ts (hunk 8 above)
Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BTable/table.d.ts]: I suppose I'm not opposed to placing this file in the composables folder. However, if its moved, it should likely receive a more specific name. The explicit use of the entire props object is an issue for me, I find it very messy. If possible, perhaps it would be better to instead split them into many smaller (and more readable) composables. (Assuming that's doable and the current composable can be 'decomposed')

**Comment by VividLemon** [comment]
You're going to need to either clean up some of these props or define what they're actually used for. A few weeks ago, I had done some cleanup as your code mimicked a "passive vmodel". If you define what some of the props you made during your last dive into the table, perhaps we can discuss if they're actually needed, or better methods that can be done

**Comment by VividLemon** [comment]
Also you should review https://github.com/orgs/bootstrap-vue-next/projects/1/views/1?pane=issue&itemId=29332903 as that is the direction that I'd like to push BTable to be in v1

**Comment by devhus** [comment]
> You're going to need to either clean up some of these props or define what they're actually used for. A few weeks ago, I had done some cleanup as your code mimicked a "passive vmodel". If you define what some of the props you made during your last dive into the table, perhaps we can discuss if they're actually needed, or better methods that can be done.
> 
> In this instance, at first glance, I'm curious about the effect of sort internal.

Almost all props have a very good use, i don't think there is any props that can be removed tbh.
For the ``sort-internal`` prop it is useful when you are sorting the table from a paginated server-request.
when ``sort-internal`` is true, the table will sort the displayed items 
when ``sort-internal`` is false, the table will not sort the displayed items but instead will only emit sort props to the given ``sorted`` event so you can request an updated item list from your server and update the ``:items`` ref based on your response.
Think about ``sort-internal`` = false as a simpler alternative to the ``provider`` prop, it's quite useful and i am using it in 2 production projects.

Anyway, I was planning to document the props, emits, and most of the BTable code so it can be understood better. Maybe i will make the component's Doc page when i get some more free time.

**Comment by devhus** [comment]
> Also you should review https://github.com/orgs/bootstrap-vue-next/projects/1/views/1?pane=issue&itemId=29332903 as that is the direction that I'd like to push BTable to be in v1

Should be easy to do but not in this PR because it is already 5 AM in my country. I will consider these changes with the next PR with the rest of the bug fixes.
For now, this PR is wrapped and will not have any more updates, But feel free to tell me if any of the previous commits need to be updated to accept the PR.

**Comment by devhus** [comment]
@VividLemon all tests now are passed, goodnight

**Comment by devhus** [comment]
@VividLemon BTableLite is created and, BTable is now a wrapper for BTableLite.

**Comment by VividLemon** [comment]
https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/1262/commits/480ac8b09e56a27aa360aca4b1909d09a101a109

Review https://vueuse.org/core/useVModel/ , specifically, the type definitions section. To clarify, the "passive" option, by using a "passive" vmodel, you aren't required to pass in a prop in order to have it work. You can modify the value like it is bound, and use it like it is bound, but not require it. For instance, in your old code you used a variable called "internalBusyFlag". This value mimiced the idea of a passive vmodel, in that it is used as if a value is bound, and changes the behavior internally as if exists.

So, long story short, use a passive vmodel. 

However, this leads to a bigger question on what the behavior of sorting is if you don't specify a sortBy

**Comment by devhus** [comment]
@VividLemon very good to know.

when using a shell value or the ``passive`` option on useVModel they will behave the same way.
Simply the sorting will still work in all causes but the given ``ref`` will not updated unless used with ``v-model:sort-by`` and in the video below you can see how the table behaves when using ``v-model:sort-by``, `:sort-by`, and removing the prop.

https://www.loom.com/share/6f22865acfc6480ba04d738863362163?sid=c6a0de79-f683-440b-a588-61bc10a83fca

**Comment by VividLemon** [comment]
> @VividLemon very good to know, updated the code to use `passive` mode.
> 
> when using a shell value or the `passive` option on useVModel they will behave the same way. Simply the sorting will still work in all causes but the given `ref` will not updated unless used with `v-model:sort-by` and in the video below you can see how the table behaves when using `v-model:sort-by`, `:sort-by`, and removing the prop.
> 
> https://www.loom.com/share/6f22865acfc6480ba04d738863362163?sid=c6a0de79-f683-440b-a588-61bc10a83fca

Passive behavior is something that I didn't know about either, but comes in great handy when working with libraries. In addition to this, when you had initially developed the internalBusyFlag and other improvements, we did not use vueuse and couldn't take advantage of this feature.

**Comment by devhus** [comment]
@VividLemon btw these are the features that currently exist in BTableLite:
- basic controlling like (foot, sticky, stacked.. etc)
- selection
- busy
- row details
- pagination

And for BTable it includes all the features of BTable ofc with extra advanced features:
- item custom providing
- sorting
- filtering,

Let me know if there are any features that need to be moved from BTableLite to BTable so i can wrap up this PR and start working on the tables docs.

**Comment by VividLemon** [comment]
> @VividLemon btw these are the features that currently exist in BTableLite:
> 
>     * basic controlling like (foot, sticky, stacked.. etc)
> 
>     * selection
> 
>     * busy
> 
>     * row details
> 
>     * pagination
> 
> 
> And for BTable it includes all the features of BTableLite ofc with extra advanced features:
> 
>     * item custom providing
> 
>     * sorting
> 
>     * filtering,
> 
> 
> Let me know if there are any features that need to be moved from BTableLite to BTable so i can wrap up this PR and start working on the tables docs.

Pagination and busy should be moved out of BTableLite. It's debatable if selection is needed on BTableLite -- although I think it would be difficult to offload that issue onto the BTable wrapper component. I'm not sure what row details does but it seems fine to include it (I believe it's in the bootstrap-vue component so that's good)

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue [hunk 11]
```diff
@@ -319,198 +163,89 @@ const emit = defineEmits<{
...
-  emit('headClicked', fieldKey, field, event, isFooter)
-
+const onFieldHeadClick = (
+  fieldKey: LiteralUnion<string>,
+  field: TableField,
+  event: MouseEvent,
+  isFooter = false
+) => {
   handleFieldSorting(field)
+  emit('headClicked', fieldKey, field, event, isFooter)
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTable.vue:233]: This wasn't directly from this PR, but all events should be kebab-case, rather than camelCase. I'm surprised I didn't notice this before. It may be worth double checking if this issue exists anywhere else (of course, `update:` prefix & modelValue for models is an exception)

File: packages/bootstrap-vue-next/src/components/BTable/BTableLite.vue [hunk 12]
```diff
@@ -0,0 +1,522 @@
...
+)
+
+const emit = defineEmits<{
+  'headClicked': [
+    key: TableFieldObject['key'],
+    field: TableField,
+    event: MouseEvent,
+    isFooter: boolean
+  ]
+  'rowClicked': [item: TableItem, index: number, event: MouseEvent]
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTableLite.vue:268]: As mentioned in a previous comment, events should be kebab case... This is likely v 0.10 worthy and is a breaking change

File: packages/bootstrap-vue-next/src/components/BTable/BTableLite.vue [hunk 13]
```diff
@@ -0,0 +1,522 @@
+<template>
+  <BTableSimple v-bind="containerAttrs">
+    <thead>
+      <slot v-if="$slots['thead-top']" name="thead-top" />
+      <tr>
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTableLite.vue:5]: Under the hood, tr, th, thead, etc etc should all use the internal components - BTr, BTh, BThead, etc etc

This will likely modify how classes in the parent are derived... As I think the only thing the wrapper components are used for are for variant classes.

**Review by VividLemon — CHANGES_REQUESTED** [review]
There's likely one more round of review to go, considering the size of the PR. But everything considered, thank you for the PR.

File: packages/bootstrap-vue-next/src/components/BTable/table-lite.spec.ts [hunk 14]
```diff
@@ -0,0 +1,555 @@
+import {enableAutoUnmount, flushPromises, mount} from '@vue/test-utils'
+import {afterEach, describe, expect, it} from 'vitest'
+import {ref} from 'vue'
```
[the remark this answers is not in the document]
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/table-lite.spec.ts:3]: I don't remember, must been a mistake or testing something out xD

[the remark this answers is not in the document]
Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BTable/table-lite.spec.ts:3]: I think it may be an unused variable. So perhaps you may want to give a quick test lint

File: packages/bootstrap-vue-next/src/components/BTable/BTableLite.vue (hunk 13 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/BTableLite.vue:5]: I will leave this update to the next PR because it's 6 AM here and gotta go, And will be too busy to update this PR in the upcoming days. So let's wrap it up to let you be free to publish it.

**Comment by devhus** [comment]
@VividLemon I did all that can be done for now, Feel free to publish the PR or ask for more reviews to be handled if my time allows it.
Goodnight.

**Comment by VividLemon** [comment]
> @VividLemon I did all that can be done for now, Feel free to publish the PR or ask for more reviews to be handled if my time allows it. Goodnight.

👋 I'll review it tomorrow, and assuming its good, it will be published. I'll make an issue tomorrow about the event names so it can be tracked.

**Comment by devhus** [comment]
@VividLemon check out the Docs layout updates.
It does feel more like Bootstrap V5 styling to me now, let me know what you think.

Before:
image

After:
image

**Comment by VividLemon** [comment]
Navbar dark is deprecated

**Comment by devhus** [comment]
@VividLemon will update it.

i've changed a few things:
- navbar from fluid-container to container
- removed borders of the content sidebars 
- adjusted a few white spaces
- add active state to the left sidebar 
- added active state to the navbar links as well
image
- fixed sidebar toggler color on mobile view, it's now working correctly in the dark and light mode unlike before
- fixed spaces of nav elements on mobile view 

Before
image

After
image

File: apps/docs/.vitepress/theme/Layout.vue [hunk 15]
```diff
@@ -204,6 +204,7 @@ import {useData, useRoute, withBase} from 'vitepress'
 import {appInfoKey} from './keys'
 import {useMediaQuery} from '@vueuse/core'
 import TableOfContentsNav from '../../src/components/TableOfContentsNav.vue'
+import {vBColorMode} from 'bootstrap-vue-next'
```
Comment (VividLemon) [inline apps/docs/.vitepress/theme/Layout.vue:207]: You duplicated this import

Comment (devhus) [reply apps/docs/.vitepress/theme/Layout.vue:207]: It's not working without importing it.

File: apps/docs/.vitepress/theme/Layout.vue [hunk 16]
```diff
@@ -1,79 +1,118 @@
...
-        <title>BootstrapVueNext</title>
-        <path
-          fill="currentColor"
-          d="M92 0h769c50 0 92 42 92 92v769c0 50-42 92-92 92H92c-50 0-92-42-92-92V92C0 42 42 0 92 0zm216 710c100 0 160-50 160-133 0-62-44-107-108-113v-3c48-8 86-52 86-102 0-71-55-117-140-117H111v468h197zM195 307h90c50 0 78 23 78 64 0 44-33 68-91 68h-77V307zm0 338V499h90c64 0 98 25 98 73s-33 73-94 73h-94zm503 65l163-468h-90L652 621h-2L531 242h-92l163 468h96z"
-        />
-      </svg>
-    </BNavbarBrand>
+  <BNavbar variant="primary" sticky="top" toggleable="lg" :container="true" v-b-color-mode="'dark'">
+    <div class="d-flex gap-2 align-items-center">
+      <BNavbarToggle v-b-toggle.sidebar-menu class="text-light" />
```
Comment (VividLemon) [inline apps/docs/.vitepress/theme/Layout.vue:4]: Are you sure you need text-light if the color mode of the parent is dark?

File: apps/docs/.vitepress/theme/Layout.vue [hunk 17]
```diff
@@ -1,79 +1,118 @@
...
+    </div>
     <BCollapse is-nav>
       <BNavbarNav>
         <BNav>
-          <BNavItem :to="withBase('/docs')">Getting Started</BNavItem>
-          <BNavItem :to="withBase('/docs/icons')">Icons</BNavItem>
-          <BNavItem :to="withBase('/docs/types')">Types</BNavItem>
-          <BNavItem :to="withBase('/docs/reference')">Reference</BNavItem>
-          <BNavItem :to="withBase('/docs/migration-guide')">Migrate</BNavItem>
+          <BNavItem
```
Comment (VividLemon) [inline apps/docs/.vitepress/theme/Layout.vue:30]: It wasn't necessarily built this way, but you may prefer to create a variable in the script and use a v-for. Rather than duplicate these similar components a few times.

File: apps/docs/.vitepress/theme/Layout.vue [hunk 18]
```diff
@@ -1,79 +1,118 @@
...
-          </BDropdownItem>
-        </BNavItemDropdown>
-      </ClientOnly>
-    </BNav>
-    <BNavbarToggle v-b-toggle.otp-menu class="otp-menu-toggle">
-      On the page <ChevronRight aria-hidden />
-    </BNavbarToggle>
+    <div class="d-flex gap-2 flex-wrap">
+      <BNav class="d-flex gap-4">
+        <BNavItem
```
Comment (VividLemon) [inline apps/docs/.vitepress/theme/Layout.vue:70]: This could probably also do the above comment

File: apps/docs/.vitepress/theme/Layout.vue [hunk 19]
```diff
@@ -1,79 +1,118 @@
...
+              <component :is="currentIcon" :aria-label="`Toggle theme (${dark})`" />
+            </template>
+            <BDropdownItem v-for="el in options" :key="el" :active="dark === el" @click="set(el)">
+              <component :is="map[el]" /> {{ el }}
+            </BDropdownItem>
+          </BNavItemDropdown>
+        </ClientOnly>
+      </BNav>
+      <BNavbarToggle v-b-toggle.otp-menu class="otp-menu-toggle">
+        On the page
```
Comment (VividLemon) [inline apps/docs/.vitepress/theme/Layout.vue:112]:
```suggestion
        On this page
```
Is this correct?

File: apps/docs/.vitepress/theme/Layout.vue (hunk 15 above)
Comment (VividLemon) [reply apps/docs/.vitepress/theme/Layout.vue:207]: Move this 
image

File: apps/docs/src/components/TableOfContentsNav.vue [hunk 20]
```diff
@@ -1,50 +1,68 @@
 <template>
   <div class="bd-links-nav">
     <BListGroup v-if="!isLargeScreen">
-      <strong class="bd-links-heading"><GearIcon aria-hidden /> General</strong>
+      <strong class="bd-links-heading"> <GearIcon aria-hidden /> General </strong>
       <BListGroupItem>
-        <BLink :to="withBase('/docs')">Getting Started</BLink>
+        <BLink
```
Comment (VividLemon) [inline apps/docs/src/components/TableOfContentsNav.vue:6]: This series of components could probably do the above suggestion

File: apps/docs/src/components/TableOfContentsNav.vue [hunk 21]
```diff
@@ -151,4 +171,25 @@ const directivesComputedList = computed(() =>
     }))
     .sort((a, b) => a.name.localeCompare(b.name))
 )
+
+const groupComputedList = computed(() => [
+  {
+    label: 'Components',
+    uri: '/docs/components',
+    icon: () => IntersectIcon,
+    children: componentsComputedList.value,
```
Comment (VividLemon) [inline apps/docs/src/components/TableOfContentsNav.vue:180]: I originally made these things computeds, but they don't seem to be reactive. You may be able to declare all of the computeds here as regular, non-reactive variables. If you do that, you likely don't need to make the icon property a function as there shouldn't be any reactivity overhead (perf)

File: packages/bootstrap-vue-next/src/components/BLink/BLink.vue [hunk 22]
```diff
@@ -153,6 +153,7 @@ const routerAttr = computed(() => ({
 
 const computedLinkClasses = computed(() => ({
   active: activeBoolean.value,
+  [props.activeClass ?? '']: activeBoolean.value,
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BLink/BLink.vue:156]: activeClass is used above. Which indicates that there was some issue somewhere. This may warrant some addition investigation on how Bootstrap-vue did this. Do you mind doing some investigation on this for me?

File: packages/bootstrap-vue-next/src/components/BNav/BNavItem.vue [hunk 23]
```diff
@@ -4,7 +4,7 @@
       class="nav-link"
       :class="linkClasses"
       v-bind="{...linkProps, ...linkAttrs}"
-      active-class="active"
+      :active-class="activeClass ?? 'active'"
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BNav/BNavItem.vue:7]: props.activeClass will never be undefined by definition

File: packages/bootstrap-vue-next/src/components/BLink/BLink.vue [hunk 24]
```diff
@@ -38,7 +38,7 @@ defineSlots<{
 
 const props = withDefaults(defineProps<BLinkProps>(), {
   active: undefined,
-  activeClass: 'router-link-active',
+  activeClass: undefined,
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BLink/BLink.vue:41]: Why was this changed

File: packages/bootstrap-vue-next/src/composables/useTableItems.ts [hunk 25]
```diff
@@ -0,0 +1,158 @@
...
+  perPage?: number
+  currentPage?: number
+  filter?: string
+  filterable?: string[]
+  sortBy?: string
+  sortDesc?: Booleanish
+  sortCompare?: BTableSortCompare
+}
+
+export const useTableItems = (
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/composables/useTableItems.ts:16]: Only export this as default. Using a named export here in addition to a default export is redundant

File: packages/bootstrap-vue-next/src/composables/useTableItems.ts [hunk 26]
```diff
@@ -0,0 +1,158 @@
...
+    internalItems,
+    computedItems,
+    updateInternalItems,
+    filteredHandler,
+    notifyFilteredItems,
+    computedDisplayItems,
+  }
+}
+
+const mapItems = (
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/composables/useTableItems.ts:81]: These should probably go above the exported item. I'm surprised it didn't error out with 'no use before define', but the `export` likely solves that issue.

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue [hunk 27]
```diff
@@ -556,75 +357,126 @@ const callItemsProvider = async () => {
...
-    : undefined,
-  {
-    [`table-${field.variant}`]: field.variant !== null,
-    'b-table-sticky-column': field.stickyColumn,
-  },
-]
+const getRowClasses = (
+  item: TableItem | null,
+  type: string
+): string | Array<any> | null | undefined => {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTable.vue:369]:
```suggestion
): string | any[] | null | undefined => {
```
Use [] syntax over Array syntax

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue [hunk 28]
```diff
@@ -698,3 +548,4 @@ defineExpose({
   refresh: callItemsProvider,
 })
 </script>
+./tableItems ../../composables/tableItems
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTable.vue:551]: What is this. How did this not error immediately? lol

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue [hunk 29]
```diff
@@ -685,9 +537,7 @@ watch(
 )
 
 onMounted(() => {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTable.vue:539]: Technically you can make this now
`onMounted(callItemsProvider)`

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: apps/docs/src/components/TableOfContentsNav.vue (hunk 21 above)
Comment (devhus) [reply apps/docs/src/components/TableOfContentsNav.vue:166]: I know the icon doesn't need to be a function, but it prevents others from having issues when converting the variable to a reactive variable if they ever do.
So it will stay as a function because there is no harm from doing so

File: packages/bootstrap-vue-next/src/composables/useTableItems.ts (hunk 26 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/composables/useTableItems.ts:81]: That's what a call a "God blessing that avoids the bomb"

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue (hunk 28 above)
Comment (devhus) [reply packages/bootstrap-vue-next/src/components/BTable/BTable.vue:551]: again, a God blessing that avoids the bomb

**Comment by devhus** [comment]
@VividLemon that should do it, I don't think any more updates will be added to this PR (except fixes). Feel free to review and publish when you can so i can start writing the Docs.

**Comment by VividLemon** [comment]
@devhus an issue that I noticed recently on the table component was the lack of customization for the selectable feature in the btable component.

The issue is that youre not able to modify the selectable list from outside the component, thus you're also unable to set default selectables afaik. This is drawback that bootstrap-vue suffers from. My idea: replace the isSelected list with a passive vmodel. This way we can modify the selected list externally, and also set defaults.

I've made this issue https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1310
